test/index.test.js

```diff
@@ -556,6 +553,60 @@ describe('config parser', () => {
                     assert.deepEqual(data, JSON.parse(loadData('basic-job-with-provider.json')));
                 }));
         });
+
+        describe('pipeline templates', () => {
+            const templateTagMock = {
+                version: '1.0.0'
+            };
+            let defaultPipelineTemplate;
+
+            beforeEach(() => {
+                defaultPipelineTemplate = JSON.parse(loadData('pipeline-template.json'));
+                pipelineTemplateVersionFactoryMock.getWithMetadata.resolves(defaultPipelineTemplate);
+                pipelineTemplateTagFactoryMock.get.resolves(templateTagMock);
+            });
+
+            it('flattens basic pipeline template successfully', () =>
+                parser({
+                    yaml: loadData('pipeline-template-basic.yaml'),
+                    templateFactory: templateFactoryMock,
+                    triggerFactory,
+                    pipelineTemplateTagFactory: pipelineTemplateTagFactoryMock,
+                    pipelineTemplateVersionFactory: pipelineTemplateVersionFactoryMock
+                }).then(data => {
+                    assert.deepEqual(data, JSON.parse(loadData('pipeline-template-basic-result.json')));
+                }));
+
+            it('flattens pipeline template with shared setting', () => {
+                const pipelineTemplateMock = JSON.parse(loadData('pipeline-template-with-shared-setting.json'));
+
+                pipelineTemplateVersionFactoryMock.getWithMetadata.resolves(pipelineTemplateMock);
+
+                return parser({
+                    yaml: loadData('pipeline-template-with-shared-setting.yaml'),
+                    templateFactory: templateFactoryMock,
+                    triggerFactory,
+                    pipelineTemplateTagFactory: pipelineTemplateTagFactoryMock,
+                    pipelineTemplateVersionFactory: pipelineTemplateVersionFactoryMock
+                }).then(data => {
+                    assert.deepEqual(data, JSON.parse(loadData('pipeline-template-with-shared-setting-result.json')));
+                });
+            });
+
+            it('returns error if pipeline template does not exist', () => {
```

Comment:
Add tests to cover scenario when forbidden fields (Ex: `jobs`, `parameters`, `stages`) haven been specified in the pipeline configuration which uses pipeline template